crates/bounded_planet_core/src/math/screen_transformations.rs

```diff
@@ -0,0 +1,24 @@
+use bevy::prelude::*;
+
+/// Convert a screen point into a NDC (Normalized Device Coordinate) point.
+/// 
+/// Uses screen size as part of the conversion math.
+pub fn screen_to_ndc(point: Vec2, screen_size: Vec2) -> Vec3 {
```

Comment:
If you wanted to be really fancy you could return a `NDC` newtype wrapper around `vec3` here...